x/posts/internal/keeper/keeper.go

```diff
@@ -195,6 +195,72 @@ func (k Keeper) GetPostsFiltered(ctx sdk.Context, params types.QueryPostsParams)
 	return filteredPosts
 }
 
+func (k Keeper) getAnswersStoreKey(postID types.PostID) []byte {
+	return []byte(types.PollAnswersStorePrefix + postID.String())
+}
+
+// Save the userAnswersDetails associated with the given postID inside the current context
+// It assumes that the post exists and has a Poll inside it.
+// If userAnswersDetails are already present, the old ones will be overridden.
+func (k Keeper) SavePollUserAnswers(ctx sdk.Context, postID types.PostID, userPollAnswers types.AnswersDetails) {
+	store := ctx.KVStore(k.StoreKey)
+
+	usersAnswersDetails := k.GetPostPollAnswersDetails(ctx, postID)
+
+	if usersAnswersDetails, appended := usersAnswersDetails.AppendIfMissingOrIfUsersEquals(userPollAnswers); appended {
+		store.Set(k.getAnswersStoreKey(postID), k.Cdc.MustMarshalBinaryBare(&usersAnswersDetails))
+	}
+
+}
+
+// GetAllPostPollsAnswers returns the list of all the post polls answers associated with the given postID that are stored into the current state.
+func (k Keeper) GetPostPollAnswersDetails(ctx sdk.Context, postID types.PostID) types.UsersAnswersDetails {
```

Comment:
I would rename this to `GetPollAnswers` instead to make it more easy to comprehend